models/settings_helpers.go

```diff
@@ -44,6 +47,29 @@ func GetSettings(q reform.DBTX) (*Settings, error) {
 func SaveSettings(q reform.DBTX, s *Settings) error {
 	s.fillDefaults()
 
+	for _, pair := range []struct {
+		dur  time.Duration
+		name string
+	}{
+		{dur: s.MetricsResolutions.HR, name: "hr"},
+		{dur: s.MetricsResolutions.MR, name: "mr"},
+		{dur: s.MetricsResolutions.LR, name: "lr"},
+	} {
+		if pair.dur < time.Second {
+			return status.Error(codes.InvalidArgument, pair.name+": minimal resolution is 1s")
+		}
+		if pair.dur.Truncate(time.Second) != pair.dur {
+			return status.Error(codes.InvalidArgument, pair.name+": should be a natural number of seconds")
+		}
+	}
+
+	if s.QAN.DataRetention < 24*time.Hour {
+		return status.Error(codes.InvalidArgument, "data_retention: minimal resolution is 24h")
+	}
+	if s.QAN.DataRetention.Truncate(24*time.Hour) != s.QAN.DataRetention {
+		return status.Error(codes.InvalidArgument, "data_retention: should be a natural number of days")
```

Comment:
why don't we just use days?

Reply:
https://github.com/percona/pmm-managed/pull/245#discussion_r313765106